### DISCUSSION TITLE
How can I manually edit a path in the database?

### DISCUSSION THREAD

**Discussion by ghost** [stashapp/stash#2519] [Q&A] [answered]
How can I manually edit the paths in the database if I change the drive name?

Please add it to the [wiki](https://github.com/stashapp/stash/wiki/Manually-Editing-the-Stash-Sqlite3-database).

**Comment by Belleyy** [comment]
- You can use a sqlite editor (DB Browser for SQLite)
- Use the command to edit it (check UPDATE statement)

Anyways you shouldn't need to edit manually the path. 
If you changed the location of your file, make sure to have the new directory in your Library (Settings > Library > Add Directory) then just do a Scan. Stash will check the hash of your file and will update the path if needed.

**Comment by ghost** [comment]
It would be like this according to https://stackoverflow.com/questions/10532656/sqlite-replace-part-of-a-string
```
UPDATE galleries
SET path = replace(path, '/run/media/user/Z/Stash', '/run/media/user/X/stash')
WHERE path LIKE '/run/media/user/Z/Stash%'
```
Repeat for images and scenes.

Rescanning files is ok when there are only a few files. But this way is more efficient.

**Comment by Belleyy** [comment, marked as answer]
yea should be fine. You can add a WHERE (not really necessary)
```
UPDATE galleries SET path = replace(path, '/run/media/user/Z/Stash', '/run/media/user/X/stash' ) WHERE path LIKE '/run/media/user/Z/Stash/%' 
```
edit: well you added the where part in your edit. (but the WHERE is incorrect in your code, WERE -> WHERE)

**Comment by ghost** [comment]
How would I do this skipping errors like `UNIQUE constraint failed: scenes.path`?

**Reply by ALonelyJuicebox** [reply]
That's a defined rule within the SQLite database to avoid scenarios where you have two files with the same path in the database. You can't skip that, and it's a good thing it's there.
